grid_map_loader/src/GridMapLoader.cpp

```diff
@@ -61,9 +68,12 @@ bool GridMapLoader::load()
 
 void GridMapLoader::publish()
 {
-  grid_map_msgs::msg::GridMap message;
-  grid_map::GridMapRosConverter::toMessage(map_, message);
-  publisher_->publish(std::make_unique<grid_map_msgs::msg::GridMap>(message));
-  rclcpp::sleep_for(std::chrono::nanoseconds(duration_.nanoseconds()));
+  auto message = grid_map::GridMapRosConverter::toMessage(map_);
+  publisher_->publish(std::move(message));
+
+  if (durationInSec > 0) {
```

Comment:
Not super clear on this - why is there a sleep here at all? Do we understand that?

Reply:
See above for details. In short, keep this node running for late joining nodes to get the published message.